### ISSUE TITLE
Make documentation easier to access

### ISSUE THREAD

**Issue by matthewfcarlson** [tianocore/edk2-pytool-extensions#98]
The documentation should be accessible by web.

**Comment by matthewfcarlson** [comment]
This is in progress

**Comment by Javagedes** [comment]
Starting the documentation fix.
Steps:
1. ~~Put all documentation under the google convention and adds a pydocstyle check to the pipeline~~ #325
2. Add automatic doc generation capability
3. setup release-based documentation generation and hosting - Github.io or Readthedocs?

**Comment by Javagedes** [comment]
Step 1 Completed via #325

**Comment by spbrogan** [comment]
Removing from 0.20.0 as we want to make that release and task 2 and 3 are not yet complete (as far as i know).  We can catch this in the next release.

**Comment by Javagedes** [comment]
Correct. 1 was just finished in both edk2-pytool-library and edk2-pytool-extensions. 2 is almost done.

**Comment by Javagedes** [comment]
Number 2 is completed via #348 

I still need to complete Number 2 for edk2-pytool-library, then number 3 will be quick. Planning to upload to readthedocs.

**Comment by Javagedes** [comment]
Added check to verify mkdocs builds with no warning or error in #357

**Comment by Javagedes** [comment]
**For edk2-pytool-extensions**: Documentation** generation is working and there is an automatic github action to push updated documentation to github pages on a release.

Moving this task to edk2-pytool-library to continue tracking this issue on that side.
